### PR TITLE
Enabled Get Apps in GitOps mode and other changes to Apps GitOps

### LINKED ISSUE #3700: Implement jx get apps for GitOps mode.
### Summary 

The command `jx get apps` is currently not showing any installed apps when using GitOps mode.

### Steps to reproduce the behavior

Create a cluster with `--ng` or `--gitops` flags.
Add an app with `jx add app`
List them with `jx get apps`

Notice no apps are shown.

### Expected behavior
There should be a list of all installed applications showing up.

### Actual behavior

No apps being shown after adding them in GitOps mode and executing `jx get apps`

### Jx version

The output of `jx version` is:

```
2.0.0-dev+c02375dc7 (Master)
```

### Kubernetes cluster

`jx create cluster gke --ng`

### LINKED ISSUE #3734: jx get apps ignores environment repo
### Summary 

The `GetApps()` method on `InstallOptions` in `install.go` ignores gitops mode entirely, just looking in the cluster for the running apps. The list of running app CRDs is not really the source of truth in this case, the git repo is. We should be taking the list of apps from the git repo, and then adding the status from the cluster. For example if the app is in the git repo but not yet running the state should be "installing" etc.

Until we do this the support for prefixed names in `get apps` and `delete app` won't work (some tests are there, disabled, and some code commented out).

### PR DESCRIPTION
**Pull request by dgozalo**
Added a way to override an app's `app.yaml` file to add custom values and enabled GitOps for Get Apps

feature: the app.yaml file shouldn't have the name of the app as a prefix

feature: removed unused unit tests, refactored, created new test for get apps in GitOps mode

chore: fix fmt

chore: removed a TODO

chore: fixed tests
Signed-off-by: Daniel Gozalo <dgozalo@cloudbees.com>

#### Submitter checklist

- [X] Change is code complete and matches issue description.
- [X] Change is covered by existing or new tests.

#### Description
Enabled Get Aps in GitOps mode, the functionality is as follows:
When in GitOps mode
1. Find the environment's repository
2. Perform a checkout of the Master branch
3. Check the `requirements.yaml` file for any app (matching "jx-app-*"
4. Check if it has been deployed by querying the Apps CRD
5. Exists: The app can be deployed in N namespaces, show all of them
6. Doesn't exist: Return the data in the `app.yaml` file
7. If it has a namespace, it has been deployed, thus show a "DEPLOYED" status
8. If it doesn't have a namespace, show an alternative status.

-----
Also added a way to modify the `app.yaml` file of an App when applying the environment in GitOps mode. This way we can add custom metadata to the already existing `app.yaml` file. In this case, extra labels to be able to filter through the Apps CRD.

#### Which issue this PR fixes

fixes #3700
fixes #3734

### CODE REVIEW

File: pkg/apps/gitops.go [hunk 1]
```diff
@@ -128,10 +132,11 @@ func (o *GitOpsOptions) DeleteApp(app string, alias string) error {
 		}
 		return nil
 	}
+	appGitName := strings.TrimPrefix(app, "jx-app-")
```
Comment (pmuir) [inline pkg/apps/gitops.go]: It looks like here you're removing the `jx-app` prefix from the delete command only?

Comment (dgozalo) [reply pkg/apps/gitops.go]: Fixed.

File: pkg/apps/gitops.go [hunk 2]
```diff
@@ -148,3 +150,68 @@ func (o *GitOpsOptions) DeleteApp(app string, alias string) error {
...
+	err = yaml.Unmarshal(requirementsFile, &reqs)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't unmarshal the environment's requirements.yaml file")
+	}
+
+	appsList := v1.AppList{}
+	for _, d := range reqs.Dependencies {
+		if appNames[d.Name] == true || len(appNames) == 0 {
+			//Make sure we only consider apps
+			if match, _ := regexp.MatchString("jx-app-.*", d.Name); match {
```
Comment (pmuir) [inline pkg/apps/gitops.go]: Sorry, didn't spot this before, but we can't do this as there is no requirement for an app to be called `jx-app-XXX`, that's just a convention. Every requirement except `jenkins-x-platform` is an app right now.

File: pkg/apps/install.go [hunk 3]
```diff
@@ -146,25 +146,33 @@ func (o *InstallOptions) AddApp(app string, version string, repository string, u
...
+		opts := GitOpsOptions{
+			InstallOptions: o,
+		}
+		return opts.GetApps(appsMap, o.getAppsFromCRDAPI)
 	}
+	return o.getAppsFromCRDAPI(in)
+
+}
+
+func (o *InstallOptions) getAppsFromCRDAPI(appNames []string) (*jenkinsv1.AppList, error) {
```
Comment (pmuir) [inline pkg/apps/install.go]: Might be neater to move this to the `helmops.go`?

File: pkg/jx/cmd/cmd_test_helpers/app_test_helpers.go [hunk 4]
```diff
@@ -275,3 +279,120 @@ func CreateAppTestOptions(gitOps bool, t *testing.T) *AppTestOptions {
 	return &o
 
 }
+
+// CreateAppTestOptionsForApp configures the mock environment for running apps related tests for a given App
```
Comment (pmuir) [inline pkg/jx/cmd/cmd_test_helpers/app_test_helpers.go]: This seems to repliacate a lot of `CreateAppTestOptions` - can we just pass this in as an option to the other function, and ignore if not set?

File: pkg/jx/cmd/step_helm_apply.go [hunk 5]
```diff
@@ -274,6 +279,50 @@ func (o *StepHelmApplyOptions) applyTemplateOverrides(chartName string) error {
 	return err
 }
 
+func (o *StepHelmApplyOptions) applyAppsTemplateOverrides(chartName string) error {
```
Comment (pmuir) [inline pkg/jx/cmd/step_helm_apply.go]: This could probably do with a test

**Comment by hferentschik** [comment]
Where do we stand on this one? Seems it needs some rebasing and some update to the commit messages (DCO)

**Comment by dgozalo** [comment]
> Where do we stand on this one? Seems it needs some rebasing and some update to the commit messages (DCO)

Ah, the conflicts should have appeared in a recent merge, it was fine yesterday. I'll fix it.

**Pull request closed without merging**